Changed call to unicode() to work with Python 3.

paulosalem opened the pull request:
Python 3 no longer supports the unicode() function. I got this error when I tried the gv plugin:

    NameError: name 'unicode' is not defined

It seems that str() now does the job, so I selectively (i.e., when Python version is >= 3) changed calls to unicode() to str(). After I made the change, it worked with Python 3, I got my drawings. However, I have not tested it in Python 2 (but it should work).

More details about the unicode() function here: https://stackoverflow.com/questions/36110598/nameerror-name-unicode-is-not-defined

paulosalem committed:
Changed call to unicode() to work with Python 3.

Python 3 no longer supports the unicode() function. I got this error when I tried the gv plugin:

    NameError: name 'unicode' is not defined

It seems that str() now does the job, so I selectively (i.e., when Python version is >= 3) changed calls to unicode() to str(). After I made the change, it worked with Python 3, I got my drawings. However, I have not tested it in Python 2 (but it should work).

More details here about the unicode() function here: https://stackoverflow.com/questions/36110598/nameerror-name-unicode-is-not-defined

fpom commented:
Thanks for the patch! I've fixed it another way: SNAKES has a dedicated module snakes.compat to handle such differences between Python versions. So I've added unicode to it instead of changing plugin gv.

Fixed by 22e40320842dd8b7917f22cac925029f4cd2e278

paulosalem commented:
You are welcome. Great work, thank you for this lib!

Best,

--
Dr. Paulo Salem

[www]    www.paulosalem.com
[twitter] twitter.com/paulosalem
[phone]   (+55)(11)991797077

On Sat, Dec 9, 2017 at 1:50 PM, Franck Pommereau <notifications@github.com>
wrote:

> Thanks for the patch! I've fixed it another way: SNAKES has a dedicated
> module snakes.compat to handle such differences between Python versions. So
> I've added unicode to it instead of changing plugin gv.
>
> Fixed by 22e4032
> <https://github.com/fpom/snakes/commit/22e40320842dd8b7917f22cac925029f4cd2e278>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fpom/snakes/pull/15#issuecomment-350479655>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABoVXA7EWQIij85qmPUGmg5b8-PntRvOks5s-qxXgaJpZM4Qq9-e>
> .
>
